src/app-layer-enip-common.h

```diff
@@ -0,0 +1,250 @@
+/* Copyright (C) 2015 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Kevin Wong <kwong@solananetworks.com>
+ */
+
+#ifndef __APP_LAYER_ENIP_COMMON_H__
+#define __APP_LAYER_ENIP_COMMON_H__
+
+#include "app-layer-protos.h"
+#include "app-layer-parser.h"
+#include "flow.h"
+#include "queue.h"
+
+#define MAX_ENIP_CMD    65535
+
+// EtherNet/IP commands
+#define NOP                0x0000
+#define LIST_SERVICES      0x0004
+#define LIST_IDENTITY      0x0063
+#define LIST_INTERFACES    0x0064
+#define REGISTER_SESSION   0x0065
+#define UNREGISTER_SESSION 0x0066
+#define SEND_RR_DATA       0x006F
+#define SEND_UNIT_DATA     0x0070
+#define INDICATE_STATUS    0x0072
+#define CANCEL             0x0073
+
+//Common Packet Format Types
+#define NULL_ADDR               0x0000
+#define CONNECTION_BASED        0x00a1
+#define CONNECTED_DATA_ITEM     0x00b1
+#define UNCONNECTED_DATA_ITEM   0x00b2
+#define SEQUENCE_ADDR_ITEM      0xB002
+
+//status codes
+#define SUCCESS               0x0000
+#define INVALID_CMD           0x0001
+#define NO_RESOURCES          0x0002
+#define INCORRECT_DATA        0x0003
+#define INVALID_SESSION       0x0064
+#define INVALID_LENGTH        0x0065
+#define UNSUPPORTED_PROT_REV  0x0069
+
+#define MAX_CIP_SERVICE     127
+#define MAX_CIP_CLASS       65535
+#define MAX_CIP_ATTRIBUTE   65535
+
+// CIP service codes
+#define CIP_RESERVED        0x00
+#define CIP_GET_ATTR_ALL    0x01
+#define CIP_GET_ATTR_LIST   0x03
+#define CIP_SET_ATTR_LIST   0x04
+#define CIP_RESET           0x05
+#define CIP_START           0x06
+#define CIP_STOP            0x07
+#define CIP_CREATE          0x08
+#define CIP_DELETE          0x09
+#define CIP_MSP             0x0a
+#define CIP_APPLY_ATTR      0x0d
+#define CIP_GET_ATTR_SINGLE 0x0e
+#define CIP_SET_ATTR_SINGLE 0x10
+#define CIP_KICK_TIMER      0x4b
+#define CIP_OPEN_CONNECTION 0x4c
+#define CIP_CHANGE_START    0x4f
+#define CIP_GET_STATUS      0x50
+
+//PATH sizing codes
+#define PATH_CLASS_8BIT         0x20
+#define PATH_CLASS_16BIT        0x21
+#define PATH_INSTANCE_8BIT      0x24
+#define PATH_INSTANCE_16BIT     0x25
+#define PATH_ATTR_8BIT          0x30
+#define PATH_ATTR_16BIT         0x31 //possible value
+
+/**
+ * ENIP encapsulation header
+ */
+typedef struct ENIPEncapHdr_
+{
+    u_int64_t context;
```

Comment:
All these u_int* should be converted to uint*. The u_int* ones are not portable.